cmd/agent/main.go

```diff
@@ -13,7 +14,31 @@ import (
 	"github.com/spf13/cobra"
 )
 
-const lockFile = "/var/run/xk6-disruptor"
+const lockFile = "xk6-disruptor"
+
+// returns the path to the lock file
+func getLockPath() string {
+	// get runtime directory for user
+	lockDir := os.Getenv("XDG_RUNTIME_DIR")
+	if lockDir == "" {
+		lockDir = os.TempDir()
+	}
+
+	return path.Join(lockDir, lockFile)
+}
+
+// ensure only one instance of the agent runs
+func lockExecution() error {
+	acquired, err := process.Lock(getLockPath())
+	if err != nil {
+		return fmt.Errorf("failed to create lock file %q: %w", getLockPath(), err)
+	}
+	if !acquired {
+		return fmt.Errorf("another disruptor command is already in execution")
+	}
+
+	return nil
+}
```

Comment:
Is there any motivation for having a separate function doing this error disambiguation?

If we're not going to act on `acquired` perhaps `process.Lock` could return just an error here.

Reply:
At this point, we are not acting, but we could. I would prefer not to change this in this PR as it is an unrelated change,

Reply:
Maybe we could reconsider this, but this is an unrelated change to the purpose of this PR.